module/zfs/dsl_crypt.c

```diff
@@ -235,8 +235,6 @@ dsl_crypto_params_create_nvlist(dcp_cmd_t cmd, nvlist_t *props,
 	return (0);
 
 error:
-	if (wkey != NULL)
-		dsl_wrapping_key_free(wkey);
 	if (dcp != NULL)
```

Comment:
It looks like we can do a little more cleanup.  Specially `dcp` must always be non-NULL so we can drop the conditional.